src/appliance-onboarding-script/appliance_setup/run.py

```diff
@@ -139,12 +138,17 @@ def _populate_default_values_of_optional_fields_in_config(config):
         if Constant.LOCATION not in config:
             config[Constant.LOCATION] = _customer_details.cloud_details[Constant.LOCATION]
 
-
         # TODO: Point to documentation link here for getting valid regions.
         if not validate_region(config[Constant.LOCATION]):
             raise InvalidRegion(f"This feature is only available in these regions: {Constant.VALID_LOCATIONS}")
 
     if operation == 'onboard':
+        if config["register"]:
+            arc_add_on_retriever: ArcAddOnRetriever = ArcAddOnRetriever()
+            arc_add_on_details =  arc_add_on_retriever.retrieve_data(_customer_details.customer_resource)
```

Comment:
>arc_add_on_details [](http://example.com/codeflow?start=12&length=18)

For my knowledge, what will this response hold? Is it a boolean / some other data?

Reply:
This return None if there is no arcaddOn or returns a json object having the arc addon details. This is basically the arcaddOn get API called.